roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -180,82 +180,65 @@ def __init__(self, file_path):
+                        raise Exception('could not find any image_masks in nwbfile')
+                    if 'RoiCentroid' in ps.colnames:
+                        self._roi_locs = ps['RoiCentroid']
+                    if 'Accepted' in ps.colnames:
+                        self._accepted_list = ps['Accepted'].data[:]
+                    if 'Rejected' in ps.colnames:
+                        self._rejected_list = ps['Rejected'].data[:]
+                    self._roi_idx = np.array(ps.id.data)
+                else:
+                    raise Exception('could not find any PlaneSegmentation in nwbfile')
+
+            # Extracting stores images as GrayscaleImages:
+            if 'SegmentationImages' in ophys.data_interfaces:
+                images_container = ophys.data_interfaces['SegmentationImages']
+                if 'correlation' in images_container.images:
+                    self._image_correlation = images_container.images['correlation'].data[()]
+                if 'mean' in images_container.images:
+                    self._image_mean = images_container.images['mean'].data[()]
 
         # Imaging plane:
-        _optical_channel_exist = [i for i, e in enumerate(
-            _nwbchildren_type) if e == 'OpticalChannel']
-        if _optical_channel_exist:
-            self._channel_names = []
-            for i in _optical_channel_exist:
-                self._channel_names.append(nwbfile.all_children()[i].name)
+        if 'ImagingPlane' in self.nwbfile.imaging_planes:
+            imaging_plane = self.nwbfile.imaging_planes['ImagingPlane']
+            self._channel_names = [i.name for i in imaging_plane.optical_channel]
+
         # Movie location:
-        _image_series_exist = [i for i, e in enumerate(
-            _nwbchildren_type) if e == 'TwoPhotonSeries']
-        if not _image_series_exist:
-            self._extimage_dims = None
-        else:
-            self._raw_movie_file_location = \
-                nwbfile.all_children()[_image_series_exist[0]].external_file[:][0]
-            self._extimage_dims = \
-                nwbfile.all_children()[_image_series_exist[0]].dimension
-
-        # property name/data extraction:
-        self._property_name_exist = [
-            i for i in ps.colnames if i not in ['image_mask', 'pixel_mask']]
-        self.property_vals = []
-        for i in self._property_name_exist:
-            self.property_vals.append(np.array(ps[i].data))
-
-        #Extracting stores images as GrayscaleImages:
-        _greyscaleimages = [i for i in nwbfile.all_children() if type(i).__name__ == 'GrayscaleImage']
-        self._image_correlation = [i.data[()] for i in _greyscaleimages if 'corr' in i.name.lower()][0]
-        self._image_mean = [i.data[()] for i in _greyscaleimages if 'mean' in i.name.lower()][0]
+        if 'TwoPhotonSeries' in self.nwbfile.acquisition:
+            self._raw_movie_file_location = self.nwbfile.acquisition['TwoPhotonSeries'].external_file[()][0]
```

Comment:
this should not be part of SegmentationExtractors

Reply:
I was wanting to keep this only as an internal attribute for now, which could be useful to do provenance tracking no?
This will require a change to all classes, i'll do that in a separate pr then.